apps/src/lib/script-editor/LevelTokenDetails.jsx

```diff
@@ -0,0 +1,260 @@
+          options={this.levelKindOptions}
+          onChange={this.handleLevelTypeSelected}
+          clearable={false}
+          arrowRenderer={ArrowRenderer}
+          style={styles.levelTypeSelect}
+        />
+        {this.containsLegacyLevel() && (
+          <div>
+            <span style={styles.levelFieldLabel}>Skin</span>
+            <input
+              defaultValue={this.props.level.skin}
+              type="text"
+              style={styles.textInput}
+              ref={skinInput => (this.skinInput = skinInput)}
+              onChange={this.handleFieldChange.bind(this, 'skin')}
+            />
+            <div style={{float: 'right'}}>
+              <span style={styles.levelFieldLabel}>Video key</span>
+              <input
+                defaultValue={this.props.level.videoKey}
+                type="text"
+                style={styles.textInput}
+                ref={videoKeyInput => (this.videoKeyInput = videoKeyInput)}
+                onChange={this.handleFieldChange.bind(this, 'videoKey')}
+              />
+            </div>
+            <div style={{clear: 'both'}} />
+            <span style={styles.levelFieldLabel}>Difficulty</span>
+            <input
+              defaultValue={this.props.level.conceptDifficulty}
+              type="text"
+              style={styles.textInput}
+              ref={conceptDifficultyInput =>
+                (this.conceptDifficultyInput = conceptDifficultyInput)
+              }
+              onChange={this.handleFieldChange.bind(this, 'conceptDifficulty')}
+            />
+            <div style={{float: 'right'}}>
+              <span style={styles.levelFieldLabel}>Concepts</span>
+              <input
+                defaultValue={this.props.level.concepts}
+                type="text"
+                style={Object.assign({}, styles.textInput, {width: 320})}
+                ref={conceptsInput => (this.conceptsInput = conceptsInput)}
+                onChange={this.handleFieldChange.bind(this, 'concepts')}
+              />
+            </div>
+          </div>
+        )}
+        {this.props.level.ids.map((id, index) => (
+          <div key={id}>
+            {this.props.level.ids.length > 1 && (
+              <div>
+                <hr style={styles.divider} />
+                <span style={styles.levelFieldLabel}>Active</span>
+                <input
+                  type="radio"
+                  onChange={this.handleActiveVariantChanged.bind(this, id)}
+                  defaultChecked={id === this.props.level.activeId}
+                  style={styles.checkbox}
+                  name={`radio-${this.props.stagePosition}-${
+                    this.props.level.position
+                  }`}
+                />
+              </div>
+            )}
+            <VirtualizedSelect
+              options={this.levelKeyOptions}
+              value={id}
+              onChange={this.handleLevelSelected.bind(null, index)}
```

Comment:
The level name dropdown is pretty slow (see screenshot). one idea to optimize is to group levelKeyOptions by level type and make the levelbuilder select the level type first. Open to other ideas for optimization.